site/pages/api/createDisruption.ts

```diff
@@ -0,0 +1,132 @@
+    const associatedLink = formFields["associated-link"];
+
+    if (associatedLink) {
+        validateAssociatedLink(associatedLink, errors, "some-error-id");
+    }
+
+    const disruptionReason: MiscellaneousReason | PersonnelReason | EnvironmentReason | EquipmentReason | "" =
+        formFields["disruption-reason"] || "";
+
+    validateDisruptionReasons(disruptionReason, errors, "some-error-id");
+
+    const disruptionStartDate: Date | null = formFields["disruption-start-date"];
+    const disruptionStartTime: string = formFields["disruption-start-time"];
+    const disruptionEndDate: Date | null = formFields["disruption-end-date"];
+    const disruptionEndTime: string = formFields["disruption-end-time"];
+    const disruptionIsNoEndDateTime: string | undefined = formFields.disruptionIsNoEndDateTime;
+
+    validateDateTimeSection(
+        disruptionStartDate,
+        disruptionStartTime || "",
+        disruptionEndDate,
+        disruptionEndTime || "",
+        disruptionIsNoEndDateTime,
+        errors,
+        "some-error-id",
+    );
+    const publishStartDate: Date | null = formFields["publish-start-date"];
+    const publishStartTime: string = formFields["publish-start-time"];
+    const publishEndDate: Date | null = formFields["publish-end-date"];
+    const publishEndTime: string = formFields["publish-end-time"];
+    const publishIsNoEndDateTime: string | undefined = formFields.publishIsNoEndDateTime;
+
+    validateDateTimeSection(
+        publishStartDate,
+        publishStartTime || "",
+        publishEndDate,
+        publishEndTime || "",
+        publishIsNoEndDateTime,
+        errors,
+        "some-error-id",
+    );
+
+    const disruptionData: PageInputs = {
+        typeOfDisruption: disruptionType,
+        summary: summary,
+        description: description,
+        "associated-link": associatedLink,
+        "disruption-reason": disruptionReason,
+        "disruption-start-date": disruptionStartDate ? getDateTime(disruptionStartDate).toDate() : disruptionStartDate,
+        "disruption-end-date": disruptionEndDate ? getDateTime(disruptionEndDate).toDate() : disruptionEndDate,
+        "disruption-start-time": disruptionStartTime,
+        "disruption-end-time": disruptionEndTime,
+        disruptionIsNoEndDateTime: disruptionIsNoEndDateTime,
+        "publish-start-date": publishStartDate ? getDateTime(publishStartDate).toDate() : publishStartDate,
+        "publish-end-date": publishEndDate ? getDateTime(publishEndDate).toDate() : publishEndDate,
+        "publish-start-time": publishStartTime,
+        "publish-end-time": publishEndTime,
+        publishIsNoEndDateTime: publishIsNoEndDateTime,
+        disruptionRepeats: formFields.disruptionRepeats,
+    };
+
+    setCookieOnResponseObject(
+        COOKIES_DISRUPTION_INFO,
+        JSON.stringify(disruptionData),
+        res,
+        TEN_SECONDS_IN_MILLISECONDS,
+        false,
+    );
+
+    if (errors.length == 0) {
+        redirectTo(res, "/");
```

Comment:
I think we need a 'return' after the redirect.

Reply:
done